clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java

```diff
@@ -1016,8 +1009,24 @@ public void run() {
           pushId = Version.generateRePushId(pushId);
           if (storeSetting.sourceKafkaInputVersionInfo.getHybridStoreConfig() != null
               && pushJobSetting.rewindTimeInSecondsOverride == NOT_SET) {
-            pushJobSetting.rewindTimeInSecondsOverride = DEFAULT_RE_PUSH_REWIND_IN_SECONDS_OVERRIDE;
-            LOGGER.info("Overriding re-push rewind time in seconds to: {}", DEFAULT_RE_PUSH_REWIND_IN_SECONDS_OVERRIDE);
+            // for repush with ttl, default rewind time should be 0 otherwise stale records may be re-consumed.
+            pushJobSetting.rewindTimeInSecondsOverride =
```

Comment:
The TTL is different from rewind time, and even without TTL repush, there is still a rewind time to fill the gap between the real-time write and the new version.

The rewind time should be similar for all the repushes for hybrid stores, but TTL will be very different.

One thing we could consider is to use store-level rewind time as the TTL and version-level rewind time logic should be same as today.

Reply:
> One thing we could consider is to use store-level rewind time as the TTL and version-level rewind time logic should be same as today.

IIUC, you are suggesting we use `storeSetting.sourceKafkaOutputVersionInfo.getHybridStoreConfig().getRewindTimeInSeconds()` instead of 0 when customers don't specify the `rewindTimeInSecondsOverride`?
Sure, I think this is more reasonable to gather the information from customer's store setting.